src/language-markdown/printer-markdown.js

```diff
@@ -427,6 +427,9 @@ function genericPrint(path, options, print) {
         options.locEnd(node)
       );
     }
+    case "abbr": {
+      return concat(["*[", node.abbr, "]: ", node.title.trim(), hardline]);
```

Comment:
The `hardline` between nodes are controlled by `shouldNotPrePrintHardline`, `shouldPrePrintDoubleHardline`, and `shouldPrePrintTripleHardline`, so you don't need to add it here. But in this case, I guess you could simply reuse the logic from `definition`, which looks like the same syntax with a prefix `*` added.

Reply:
I gave it another shot, but wasn't successful. I couldn't make the `definiton` logic work. I'll give it another shot later, but if you have any insight it'd be greatly appreciated.